interfaces/cython/cantera/onedim.pyx

```diff
@@ -934,6 +943,11 @@ cdef class Sim1D:
             except CanteraError as e:
                 log(str(e))
                 solved = False
+            except Exception as e:
+                # restore settings before re-raising exception
+                set_transport(self.gas, self.domains, transport)
+                set_soret(soret_doms, solve_soret)
```

Comment:
Since `set_soret` will only enable the Soret term for domains where it was originally enabled, you can just pass `True` as the argument, rather than using the boolean `solve_soret`. Then, you can use just use the Pythonic `if soret_doms:` instead of defining `solve_soret` in the first place.

Reply:
ok. My objective was clarity / symmetry to `solve_multi`, but it is indeed not necessary.